From 29de28fd125b6423253ef99c8c897a41ae92cbab Mon Sep 17 00:00:00 2001
Message-ID: <29de28fd125b6423253ef99c8c897a41ae92cbab.1764155841.git.gerald.elder-vass@citrix.com>
From: Gerald Elder-Vass <gerald.elder-vass@citrix.com>
Date: Wed, 26 Nov 2025 11:17:01 +0000
Subject: [PATCH 1/2] Revert "hw/smbios: Remove 'uuid_encoded' argument from
 smbios_set_defaults()"

This reverts commit c338128e80237ef4c8ca238940d9ffaed6211fd1.
---
 hw/arm/virt.c                | 2 +-
 hw/i386/fw_cfg.c             | 2 +-
 hw/loongarch/virt.c          | 2 +-
 hw/riscv/virt.c              | 2 +-
 hw/smbios/smbios.c           | 6 ++++--
 include/hw/firmware/smbios.h | 3 ++-
 6 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/hw/arm/virt.c b/hw/arm/virt.c
index f6bb74760f9c..35f17b96fa1b 100644
--- a/hw/arm/virt.c
+++ b/hw/arm/virt.c
@@ -1755,7 +1755,7 @@ static void virt_build_smbios(VirtMachineState *vms)
         product = "KVM Virtual Machine";
     }
 
-    smbios_set_defaults("QEMU", product, mc->name);
+    smbios_set_defaults("QEMU", product, mc->name, true);
 
     /* build the array of physical mem area from base_memmap */
     mem_array.address = vms->memmap[VIRT_MEM].base;
diff --git a/hw/i386/fw_cfg.c b/hw/i386/fw_cfg.c
index 5c0bcd5f8a9f..7709dc80a5b3 100644
--- a/hw/i386/fw_cfg.c
+++ b/hw/i386/fw_cfg.c
@@ -75,7 +75,7 @@ void fw_cfg_build_smbios(PCMachineState *pcms, FWCfgState *fw_cfg,
 
     if (pcmc->smbios_defaults) {
         /* These values are guest ABI, do not change */
-        smbios_set_defaults("QEMU", mc->desc, mc->name);
+        smbios_set_defaults("QEMU", mc->desc, mc->name, true);
     }
 
     /* tell smbios about cpuid version and features */
diff --git a/hw/loongarch/virt.c b/hw/loongarch/virt.c
index db390717a6d0..6416c2ca8842 100644
--- a/hw/loongarch/virt.c
+++ b/hw/loongarch/virt.c
@@ -140,7 +140,7 @@ static void virt_build_smbios(LoongArchVirtMachineState *lvms)
         product = "KVM Virtual Machine";
     }
 
-    smbios_set_defaults("QEMU", product, mc->name);
+    smbios_set_defaults("QEMU", product, mc->name, true);
 
     smbios_get_tables(ms, SMBIOS_ENTRY_POINT_TYPE_64,
                       NULL, 0,
diff --git a/hw/riscv/virt.c b/hw/riscv/virt.c
index 47e573f85ab1..775ba2d2f872 100644
--- a/hw/riscv/virt.c
+++ b/hw/riscv/virt.c
@@ -1402,7 +1402,7 @@ static void virt_build_smbios(RISCVVirtState *s)
         product = "KVM Virtual Machine";
     }
 
-    smbios_set_defaults("QEMU", product, mc->name);
+    smbios_set_defaults("QEMU", product, mc->name, true);
 
     if (riscv_is_32bit(&s->soc[0])) {
         smbios_set_default_processor_family(0x200);
diff --git a/hw/smbios/smbios.c b/hw/smbios/smbios.c
index 6c270d283c31..d5da53095c60 100644
--- a/hw/smbios/smbios.c
+++ b/hw/smbios/smbios.c
@@ -31,7 +31,7 @@
 #include "hw/pci/pci_device.h"
 #include "smbios_build.h"
 
-static const bool smbios_uuid_encoded = true;
+static bool smbios_uuid_encoded = true;
 /*
  * SMBIOS tables provided by user with '-smbios file=<foo>' option
  */
@@ -1018,9 +1018,11 @@ void smbios_set_default_processor_family(uint16_t processor_family)
 }
 
 void smbios_set_defaults(const char *manufacturer, const char *product,
-                         const char *version)
+                         const char *version,
+                         bool uuid_encoded)
 {
     smbios_have_defaults = true;
+    smbios_uuid_encoded = uuid_encoded;
 
     SMBIOS_SET_DEFAULT(smbios_type1.manufacturer, manufacturer);
     SMBIOS_SET_DEFAULT(smbios_type1.product, product);
diff --git a/include/hw/firmware/smbios.h b/include/hw/firmware/smbios.h
index f066ab726291..8d3fb2fb3b66 100644
--- a/include/hw/firmware/smbios.h
+++ b/include/hw/firmware/smbios.h
@@ -331,7 +331,8 @@ void smbios_add_usr_blob_size(size_t size);
 void smbios_entry_add(QemuOpts *opts, Error **errp);
 void smbios_set_cpuid(uint32_t version, uint32_t features);
 void smbios_set_defaults(const char *manufacturer, const char *product,
-                         const char *version);
+                         const char *version,
+                         bool uuid_encoded);
 void smbios_set_default_processor_family(uint16_t processor_family);
 uint8_t *smbios_get_table_legacy(size_t *length, Error **errp);
 void smbios_get_tables(MachineState *ms,
-- 
2.47.3

